### PR TITLE
Prevent users from purchasing additional mailboxes when the associated subscription expires/is expiring

### PR DESCRIPTION
**Pull request by olaseni**
#### Changes proposed in this Pull Request

* The email plan page checks the expiry status of the plan's subscription to determine if the user should be allowed to purchase additional mailboxes.
* In the event that the subscription is very close to expiry or expired, a renew button is shown instead.

#### Testing instructions

* Checkout this branch locally or use the [live branch](https://calypso.live/?branch=fix/prevent-new-mailboxes-for-expired-subs)
* View the email plan page for any email subscription i.e. Professional Email, or Google Workspace
* Confirm that you are prompted to renew when the subscription has expired
    

Screenshot 2021-06-01 at 5 00 23 AM

### CODE REVIEW

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx]: I would suggest this shorter and clearer approach:

```suggestion
					subscriptionId={ Number( purchase.id ) }
```

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx]: This label should be `Add new mailboxes` per https://github.com/Automattic/wp-calypso/pull/53161.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx]: I think `Renew to add new mailboxes` is enough but I'll leave it up to you.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx]: As far I can see `addMailboxProps` is only ever used once, so we could simply inline `this.getAddMailboxProps()`:

```javascript
			<VerticalNavItem { ...this.getAddMailboxProps() }>{ translate( 'Add new mailbox' ) }</VerticalNavItem>
```

**Review by stephanethomas — CHANGES_REQUESTED** [review]
A gentle reminder to add screenshots of the changes 🙂. This is what I see for that new option:

screenshot

It appears as a button, which feels really weird. Looking at `RenewButton` it seems we don't need all the features it provides, so I believe we could just use a regular `VerticalNavItem` and call `handleRenewNowClick()` when it is clicked.

> In the event that the subscription is very close to expiry or expired, a renew button is shown instead.

We are the May 27th today, and I was presented with this new button whereas my Titan subscription is supposed to expire on June 8th. That seems way too early to force users to renew. Looking quickly at `needsToRenewSoon()`, it seems it's even worse as we will basically show that button all the time for Titan subscriptions.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (daledupreez) [reply client/my-sites/email/email-management/home/email-plan.jsx]: Please make sure the translations are in for the other PR before merging this one with the updated string.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx:275]: `selectedSite` is always provided (it's a required prop), we can remove it from this condition.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-plan.jsx:127]: Maybe add a blank line above 🙂?

**Review by stephanethomas — APPROVED** [review]
I've tested it with a Titan subscription, and it worked great 👍.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5971840

Thank you @olaseni for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5971840).

**Pull request merged**
